### PR TITLE
chore: add centralised default error handler

### PR DESCRIPTION
**Pull request by akaJSaunders**
### JIRA

N/A

### Change Description

This PR aims to:
- centralise error handling
- check for empty sessions on rendering of pages
- add tests to make sure error is rendered on empty sessions

### Work Checklist

- [ ] Tests added where applicable
- [ ] Test coverage of new code meets target of 80%
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

---

### Merge Instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/ApplicantDetailsController.ts [hunk 1]
```diff
@@ -21,13 +21,14 @@ export class ApplicantDetailsController {
...
   public renderView = (req: Request, res: Response, next: NextFunction) => {
+
     const suppressionData = SessionService.getSuppressionSession(req);
 
     res.render(template, {
       ...this.getApplicantDetails(suppressionData),
       backNavigation
     });
-  }
+  };
```
Comment (rhysbarrett) [inline src/controllers/ApplicantDetailsController.ts:31]: Semicolons after function declarations are unnecessary.

File: src/middleware/ErrorHandler.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { NextFunction, Request, Response } from 'express';
+import { StatusCodes } from 'http-status-codes/build';
+
+// @ts-ignore
```
Comment (rhysbarrett) [inline src/middleware/ErrorHandler.ts:4]: What's being ignored here?

Comment (akaJSaunders) [reply src/middleware/ErrorHandler.ts:4]: good spot. something i left behind after a few commits. Removed it

**Review by rhysbarrett — APPROVED** [review]
